File: 0034-find-first-and-last-position-of-element-in-sorted-array/0034-find-first-and-last-position-of-element-in-sorted-array.java
class Solution {
    public int[] searchRange(int[] nums, int target) {
    int left = leftbond(nums , target);
    int right = rightbond(nums , target);      
    return new int[]{left,right};
    }
    private int leftbond(int[] nums , int target){
         int index=-1;
         int low=0;
        int high=nums.length-1;
        while(low <= high){
            int mid = low+(high-low)/2;
            if(nums[mid]==target){
                index= mid;
                high = mid-1;
            }
            else if(nums[mid] < target) low = mid+1;
            else high = mid-1;
        }
        return index;
    }
      private int rightbond(int[] nums , int target){
         int index=-1;
         int low=0;
        int high=nums.length-1;
        while(low <= high){
            int mid = low+(high-low)/2;
            if(nums[mid]==target){
                index= mid;
                low = mid + 1;
            }
             else if(nums[mid] < target) low = mid+1;
            else high = mid-1;
        }
        return index;
    }
}

File: 0034-find-first-and-last-position-of-element-in-sorted-array/0034-find-first-and-last-position-of-element-in-sorted-array.py
class Solution:

    def searchRange(self, nums, target):
        left = self.leftbond(nums, target)
        right = self.rightbond(nums, target)

        return [left, right]

    def leftbond(self, nums, target):
        index = -1
        low = 0
        high = len(nums) - 1

        while low <= high:
            mid = low + (high - low) // 2

            if nums[mid] == target:
                index = mid
                high = mid - 1

            elif nums[mid] < target:
                low = mid + 1

            else:
                high = mid - 1

        return index

    def rightbond(self, nums, target):
        index = -1
        low = 0
        high = len(nums) - 1

        while low <= high:
            mid = low + (high - low) // 2

            if nums[mid] == target:
                index = mid
                low = mid + 1

            elif nums[mid] < target:
                low = mid + 1

            else:
                high = mid - 1

        return index